src/Formik.tsx

```diff
@@ -337,42 +324,52 @@ export class Formik<Values = object, ExtraProps = {}> extends React.Component<
             handlerName: 'handleChange',
           });
         }
-        val = /number|range/.test(type)
-          ? ((parsed = parseFloat(value)), isNaN(parsed) ? '' : parsed)
-          : /checkbox/.test(type) ? checked : value;
+        value = event.target.value;
+        if (/number|range/.test(type)) {
+          const parsed = parseFloat(event.target.value);
+          value = isNaN(parsed) ? '' : parsed;
```

Comment:
How about intermediate values, e.g.
```
var value = "12."; // user is still typing the value
var num = parseFloat(value); // 12
```
With this code, 12 would be set, losing "." and impossible to type it. Or am I wrong?

This kind of type conversion should be done on blur event.

It is tricky field

Reply:
@Andreyco This is completely unrelated to this PR, as this code was not changed. It might be a separate issue that needs fixing

Reply:
I only refactored the unnecessarily terse and confusing javascript syntax here.

Reply:
still need it. please update

Reply:
@Andreyco what’s going on, can you please resolve this issue or provide some feedback.